ambry-router/src/main/java/com.github.ambry.router/EncryptJob.java

```diff
@@ -65,38 +63,34 @@
    */
   @Override
   public void run() {
-    encryptJobMetricsTracker.onJobProcessingStart();
-    ByteBuffer encryptedBlobContent = null;
-    ByteBuffer encryptedUserMetadata = null;
-    ByteBuffer encryptedKey = null;
-    Exception exception = null;
-    try {
-      if (blobContentToEncrypt != null) {
-        encryptedBlobContent = cryptoService.encrypt(blobContentToEncrypt, perBlobKey);
+    if (!isComplete()) {
+      encryptJobMetricsTracker.onJobProcessingStart();
+      ByteBuffer encryptedBlobContent = null;
+      ByteBuffer encryptedUserMetadata = null;
+      ByteBuffer encryptedKey = null;
+      Exception exception = null;
+      if (!isComplete()) {
+        try {
+          if (blobContentToEncrypt != null) {
+            encryptedBlobContent = cryptoService.encrypt(blobContentToEncrypt, perBlobKey);
+          }
+          if (userMetadataToEncrypt != null) {
+            encryptedUserMetadata = cryptoService.encrypt(userMetadataToEncrypt, perBlobKey);
+          }
+          Object containerKey = kms.getKey(accountId, containerId);
+          encryptedKey = cryptoService.encryptKey(perBlobKey, containerKey);
+        } catch (Exception e) {
+          exception = e;
+        } finally {
+          encryptJobMetricsTracker.onJobProcessingComplete();
+        }
+        completeJob(
```

Comment:
is this moved out of the `finally` on purpose (don't think it makes a difference, but curious why)